content/courses/advanced-ada/parts/data_types/records.rst

```diff
@@ -672,18 +677,149 @@ declaring. For example:
 
        type T is limited record
           E : T_Processor (T'Access);
+          --
+          --               T'Access
           --               ^^^^^^^^
           --  Per-object expression
-          --  Per-object constraint
+          --
+          --              (T'Access)
+          --              ^^^^^^^^^^
+          --   Per-object constraint
        end record;
 
     end Rec_Per_Object_Expressions;
 
 In this example, when we write :ada:`T'Access` within the declaration of the
 :ada:`T` record type, the actual value for the :ada:`Access` attribute will be
 known when an object of :ada:`T` type is created. In that sense,
-:ada:`T'Access` is a per-object expression |mdash| and a per-object constraint
-as well.
+:ada:`T'Access` is a per-object expression |mdash| :ada:`(T'Access)` is the
+corresponding per-object constraint.
+
+Note that :ada:`T'Access` is referring to the type within a type definition.
+This is generally treated as a reference to the object being created, the
+so-called *current instance*.
+
+.. todo::
+
+    - Add link to Adv_Ada_Reference_Current_Instance section ("Access Types:
+      Reference to current instance") once it's available.
+
+.. admonition:: Relevant topics
+
+   - :arm22:`3.8 Record Types <3-8>`
+
+Restrictions on per-object expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are some important restrictions on per-object expressions and per-object
+constraints:
+
+#. From all type attributes, only access attributes (:ada:`T'Access` and
```

Comment:
This is a bit confusing.  T'Access is not actually a *type* attribute, it is an attribute of the current instance *object*.